File: SDK/java/src/main/java/com/reloop/verifio/errors/VerifioException.java
package com.reloop.verifio.errors;

public class VerifioException extends RuntimeException {
    private final Integer statusCode;
    private final String code;

    public VerifioException(String message) {
        this(message, null, null);
    }

    public VerifioException(String message, Integer statusCode, String code) {
        super(message);
        this.statusCode = statusCode;
        this.code = code;
    }

    public Integer getStatusCode() {
        return statusCode;
    }

    public String getCode() {
        return code;
    }

    @Override
    public String toString() {
        if (code != null) {
            return String.format("VerifioException [%d %s]: %s", statusCode, code, getMessage());
        }
        if (statusCode != null) {
            return String.format("VerifioException [%d]: %s", statusCode, getMessage());
        }
        return "VerifioException: " + getMessage();
    }
}

File: SDK/php/src/Exceptions/VerifioException.php
<?php

namespace Verifio\Exceptions;

use Exception;

class VerifioException extends Exception
{
    protected ?int $statusCode;
    protected ?string $errorCode;

    public function __construct(string $message, ?int $statusCode = null, ?string $errorCode = null)
    {
        parent::__construct($message);
        $this->statusCode = $statusCode;
        $this->errorCode = $errorCode;
    }

    public function getStatusCode(): ?int
    {
        return $this->statusCode;
    }

    public function getErrorCode(): ?string
    {
        return $this->errorCode;
    }

    public function __toString(): string
    {
        if ($this->errorCode !== null) {
            return "VerifioException [{$this->statusCode} {$this->errorCode}]: {$this->message}";
        }
        if ($this->statusCode !== null) {
            return "VerifioException [{$this->statusCode}]: {$this->message}";
        }
        return "VerifioException: {$this->message}";
    }
}
